src/core/FreetypeRenderer.cc

```diff
@@ -299,10 +306,7 @@ void FreetypeRenderer::Params::set(Parameters& parameters)
   (void)parameters.valid("halign", Value::Type::STRING);
   (void)parameters.valid("valign", Value::Type::STRING);
 
-  set_fn(parameters["$fn"].toDouble());
-  set_fa(parameters["$fa"].toDouble());
-  set_fs(parameters["$fs"].toDouble());
-
+  discretizer = std::make_shared<CurveDiscretizer>(parameters);
```

Comment:
Generally, is it worth messing with shared_ptr on order to store 3 doubles? I can see that for small object s with lots of instances it can make a difference, but passing by value and move/copy might just be cleaner in most cases.

Reply:
I was trying to avoid header pollution. Value types require including `core/CurveDiscretizer.h` in dozens of header files instead of none. I now forget why a reference didn't work; I think it had something to do with requiring an explicitly-defined destructor in every class which uses the reference.

Reply:
Just a copy if fine too. With references you need to care about ownership and lifetimes.
I wouldn't care too much about header pollution, at least I wouldn't complicate stuff as a compile-time optimization.